File: StudentClass-OC/StudentClass-OC/Student.m
//
//  Student.m
//  StudentClass
//
//  Created by 李琪 on 2022/7/18.
//

#import "Student.h"

@implementation Student


-(id)initWithName:(NSString *)name andAge:(NSInteger)age andMajor:(NSString *)major{
    if (self = [super init]) {
        self.name = name;
        self.age = age;
        self.major = major;
    }
    return self;
}

-(NSString *)study:(float)time{
    return [NSString stringWithFormat:@"My name is %@, I am %ld years old, I specialize in %@ and I have been studying for %2.f hours",_name,(long)_age,_major,time];
}
@end

File: StudentClass-Swift/StudentClass-Swift/Student.swift
//
//  Student.swift
//  StudentClass-Swift
//
//  Created by 李琪 on 2022/7/18.
//

import Cocoa

class Student: NSObject {
    
    var name: String
    var age: Int
    var major: String
    init(name:String,age:Int,major:String) {
        self.name = name
        self.age = age
        self.major = major
    }
    func study(time:Float){
        print("My name is \(name) age is \(age) major is \(major) and I have been studying for \(time) hours")
    }
}


